no-vary-query.md

```diff
@@ -0,0 +1,337 @@
+
+- If `/worker-v2.js` contains `No-Vary-Query: "debuglog"`, but `/worker.js` does not, then `new SharedWorker('/worker.js?debuglog=1')` followed by `new SharedWorker('/worker.js?debuglog=0')` will spin up two separate workers, derived from `/worker-v2.js?debuglog=1` and `/worker-v2.js?debuglog=0` respectively.
+
+### Interaction with storage partitioning
+
+This proposal takes place fully within the bounds of the [client side storage partitioning](https://github.com/privacycg/storage-partitioning/) work. It is only about calculating the cache key within a given partition. It does not affect the computation of the partition key itself.
+
+## Alternatives considered
+
+`<link rel="canonical">` and its `Link` header counterpart could cover some of the use cases given here, if we expanded them to modify cache-matching semantics. Such expansion would likely have compatibility issues, given the large existing deployment of `<link rel="canonical">`. However, this only works for the simplest cases, such as `https://example.com/?k=v` saying that its canonical URL is `https://example.com/`, and then later being used to fulfill a request to `https://example.com/`. This does not allow a `https://example.com/?k=v1` response to fulfill a request for `https://example.com/?k=v2`, since to do so, we'd need to have previously requested `https://example.com/?k=v2` and seen that it has the same canonical URL as the stored `?k=v1` variant.
+
+For same-origin cases, it might be possible for the referrer to indicate when query parameters are insignificant (or more generally, that substituting one URL for another is acceptable). This wouldn't be workable for cross-origin uses, however, for security reasons. See the section on a future [referrer hint](#a-referrer-hint) for more on when a referrer-based variant might be useful _in addition_ to a header-based variant.
+
+The proposed solution could be given a narrower scope, such a header respected only by the preloading caches. We're open to this if necessary, but we think there's enough evidence of this being a larger problem that it's worth tackling other caches at the same time.
+
+The response header could be flipped, i.e. `Vary-Query`, so as to better match `Vary`. We chose the `No-Vary-Query` framing because it means that the "empty" default state matches existing HTTP caching semantics (i.e., that the response varies by all query parameters). A `Vary-Query` version would probably need to have different behavior when the header is absent (vary on everything) vs. when the header is present but empty (vary on the empty list, i.e. nothing), which seems awkward.
+
+The name could be `No-Vary-Search`, to better match web platform APIs. Actually, maybe we should just make this change?
+
+## Future extensions
```

Comment:
I do wonder based on the reception of this at TPAC whether we should be clearer that these are evidence that if we want these features it would be possible to extend to encompass them, as opposed to actual future extensions we anticipate implementing in the foreseeable future (at least for the first two). Especially No-Vary-Path I think has interesting implementation questions, and I'm also uneasy about regular expressions here.

If you agree, maybe we could rename this "Extensibility" and add an intro paragraph to clarify?